docs/src/catalyst_applications/nonlinear_solve.md

```diff
@@ -0,0 +1,99 @@
+
+This tutorial describes how to create `NonlinearProblem`s from Catalyst's `ReactionSystemn`s, and how to solve them using NonlinearSolve. More extensive descriptions of available solvers and options can be found in [NonlinearSolve's documentation](https://docs.sciml.ai/NonlinearSolve/stable/). If you use this in your research, please [cite the NonlinearSolve.jl](@ref nonlinear_solve_citation) and [Catalyst.jl](@ref catalyst_citation) publications.
+
+## Basic example
+Let us consider a simple dimerisation network, where a protein ($P$) can exist in a monomer and a dimer form. The protein is produced at a constant rate from its mRNA, which is also produced at a constant rate.
+```@example nonlinear_solve1
+using Catalyst
+dimer_production = @reaction_network begin
+    pₘ, 0 --> mRNA
+    pₚ, mRNA --> mRNA + P
+    (k₁, k₂), 2P <--> P₂
+    d, (mRNA, P, P₂) --> 0
+end
+```
+This system corresponds to the following ODE:
+```math
+\begin{aligned}
+\frac{dmRNA}{dt} &= pₘ - d \cdot mRNA \\
+\frac{dP}{dt} &= pₚ \cdot mRNA - k₁ \cdot P + 2k₂ \cdot P₂ - d \cdot P \\
+\frac{dP₂}{dt} &= k₁ \cdot P + 2k₂ \cdot P₂ \\
+\end{aligned}
+```
+To find its steady states we need to solve:
+```math
+\begin{aligned}
+0 &= pₘ - d \cdot mRNA \\
+0 &= pₚ \cdot mRNA - k₁ \cdot P + 2k₂ \cdot P₂ - d \cdot P \\
+0 &= k₁ \cdot P + 2k₂ \cdot P₂ \\
+\end{aligned}
+```
+
+To solve this problem, we must first designate our parameter values, and also make an initial guess of the solution. Generally, for problems with a single solution (like this one), most arbitrary guesses will work fine (the exception typically being [systems with conservation laws](@ref nonlinear_solve_conservation_laws)). Using these, we can create the `NonlinearProblem` that we wish to solve.
+```@example nonlinear_solve1
+p = [:pₘ => 0.5, :pₚ => 2.0, :k₁ => 5.0, :k₂ => 1.0, :d => 1.0]
+u_guess = [:mRNA => 1.0, :P => 1.0, :P₂ => 1.0]
+nl_prob = NonlinearProblem(dimer_production, u_guess, p)
+```
+Finally, we can solve it using the `solve` command, returning the steady state solution:
+```@example nonlinear_solve1
+using NonlinearSolve
+solve(nl_prob)
+```
+
+NonlinearSolve provides [a wide range of potential solvers](https://docs.sciml.ai/NonlinearSolve/stable/solvers/NonlinearSystemSolvers/). If we wish to designate one, it can be supplied as a second argument to `solve`. Here, we use the Newton Trust Region method:
+```@example nonlinear_solve1
+solve(nl_prob, TrustRegion())
```

Comment:
Maybe show the solutions the two methods get agree via `isapprox`?

Reply:
Makes sense, will add.